service/internal/podcidr/pod_cidr.go

```diff
@@ -38,6 +45,49 @@ func New(c Config) (*PodCIDR, error) {
 }
 
 func (p *PodCIDR) PodCIDR(ctx context.Context, obj interface{}) (string, error) {
-	// TODO Add logic.
-	return "", nil
+	cr, err := meta.Accessor(obj)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	cl, err := p.lookupCluster(ctx, cr)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	var podCIDR string
+	if cl.Spec.ClusterConfiguration == nil || cl.Spec.ClusterConfiguration.Networking.PodSubnet == "" {
```

Comment:
ClusterConfiguration is a pointer so added this nil check to fix a crash.

Reply:
isn't that what you want to?
```suggestion
	if cl.Spec.ClusterConfiguration =! nil && cl.Spec.ClusterConfiguration.Networking.PodSubnet == "" {
```

Reply:
The cluster-operator code doesn't have the nil check but the operator crashed in testing. As this block in the CR was empty.

This treats a nil ClusterConfiguration the same as the string being empty which I think is correct.